### PR TITLE
Test for resnet18 accuracy

### PR DESCRIPTION
**Pull request by kiya00**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Add test for resnet18 accuracy

2 issues: 
1. when backward+nvfuserex: https://github.com/Lightning-AI/lightning-thunder/pull/451#issuecomment-2186631228
2. when backward+torchex+fp32: numerical accuracy problem

### CODE REVIEW

**Comment by kiya00** [comment]
Hi @crcrpar , do you have any idea why backward+FP32 has problems with numerical accuracy?

File: thunder/tests/test_inplace_functionalization.py [hunk 1]
```diff
@@ -137,31 +139,52 @@ def turn_off_tf32_and_set_seed(monkeypatch):
...
 
+@instantiate(
+    dtypes=(thunder.float32, thunder.float64),
+    devicetypes=(devices.DeviceType.CUDA,),
+    decorators=(pytest.mark.parametrize("train", (False, True)),),
+)
 @requiresCUDA
-@pytest.mark.parametrize("train", (False, True))
-def test_parse_resnet18(turn_off_tf32_and_set_seed, train: bool):
+def test_parse_resnet18(executor, device, dtype, turn_off_tf32_and_set_seed, train: bool):
```
Comment (t-vi) [inline thunder/tests/test_inplace_functionalization.py:148]: It seems strange that the discrepancy is back here. Do you know if pytest is sensitive to the order here (i.e. does the fixture need to come first)?

Comment (crcrpar) [reply thunder/tests/test_inplace_functionalization.py:148]: I think it's the new discrepancy coming from the grad check

**Review by t-vi — APPROVED** [review]
Thank you @kiya00 and @crcrpar

**Comment by csarofeen** [comment]
The last comment from @kiya00 sounded like the test wasn’t passing. Was the numerical issue resolved?
